docs/_articles/en/v3.x/salesforce-graph-engine/rules.md

```diff
@@ -5,21 +5,57 @@ lang: en
+| ApexNullPointerExceptionRule | Path-based analysis | Error-Prone | GA | Identifies Apex operations that dereference null objects and throw NullPointerExceptions. |
+| AvoidDatabaseOperationInLoop | Path-based analysis | Performance | Pilot | Detects database operations in loops that degrade performance. |
+| AvoidMultipleMassSchemaLookups | Path-based analysis | Performance | GA | Detects scenarios where expensive schema lookups are made more than one time in a paths. |
+| RemoveUnusedMethod | Path-based analysis | Performance | Pilot | Detects methods contained in your code that aren’t invoked from any entry points that Graph Engine recognizes. |
+| UnimplementedTypeRule | Graph-based analysis | Performance | GA | Detects abstract classes and interfaces that are non-global and missing implementations or extensions. |
+| UseWithSharingOnDatabaseOperation | Path-based analysis | Security | Pilot | Detects database operations outside with-sharing-annotated classes. |
 
-## ApexFlsViolationRule
-ApexFlsViolationRule detects [Create, Read, Update, and Delete (CRUD) and Field-Level Security (FLS) violations](https://www.youtube.com/watch?v=1ZYjpjPTIn8). To run the path-based analysis, rules run `scanner:run:dfa`. Alternatively, run `scanner:run:dfa --category “Security”` to run only the ApexFlsViolationRule. 
+## Running Graph Engine GA Rules
+Run all Graph Engine rules against your code, or run a subset of rules by type or by category.
 
-Example: 
-```sfdx scanner:run:dfa --category "Security" --projectdir /project/dir --target /project/dir/target```
+* To run the path-based rules run: scanner:run:dfa --projectdir MyDirectory.
 
-### Definitions
+	**Example:**
+  
+	```sfdx scanner:run:dfa --projectdir /project/dir --target /project/dir/target1```
+
+* To run graph-based analysis rules run: scanner:run --engine sfge --projectdir MyDirectory.
+
+	**Example:**
+  
+	```sfdx scanner:run --engine sfge --projectdir /project/dir --target /project/dir/target1```
+
+* To run a category of rules, run: `scanner:run:dfa --category "Security"`
+  
+	**Example:**
+  
+	```sfdx scanner:run:dfa --category "Security" --projectdir /project/dir --target /project/dir/target```
+
+## Running Graph Engine Pilot Rules
+
+To run each Graph Engine pilot rule, include the ```--with-pilot``` flag in your request. 
+
+* To run all Graph Engine rules and all pilot rules, run:
+
+	```sfdx scanner:run:dfa --with-pilot --engine sfge --projectdir /project/dir --target /project/dir/target1```
+
+* To run a specific category of rules including the pilot rules in that category, include the category and the ```--with-pilot``` flag.
+
+	**Example**:
+  
+	```sfdx scanner:run:dfa --category “Performance” --with-pilot --engine sfge --projectdir /project/dir --target /project/dir/target1```
+
+## Generally Available Rules
+
+### ApexFlsViolationRule
```

Comment:
Changing this from a `##` header to a `###` header will break the URLs in the violations.
I've learned that you can manually add your own anchor with the `<a>` tag, so if you do:
`### ApexFlsViolationRule <a name='ApexFlsViolationRule'>#</a>`
then it will render like this, and the links in our violations will continue to work as expected.
<img width="325" alt="Screen Shot 2023-07-28 at 2 00 41 PM" src="https://github.com/forcedotcom/sfdx-scanner/assets/4054488/c7e5d1c3-e559-42c2-9f0c-8d462c2c15a4">

Reply:
@jfeingold35 oh dear, I wasn't aware that changing ## levels affected links. We do need to change them though since we're moving How to run rules, and How to run pilot rules up top.

Reply:
Done